app/helpers/application_helper.rb

```diff
@@ -64,40 +64,28 @@ def link_to_add_fields(name, f, association, partial = nil, options = {})
     link_to_function(name, ("add_fields(this, \"#{association}\", \"#{escape_javascript(fields)}\"); turn_textarea_switch();").html_safe, add_html_classes(options, "btn btn-success") )
   end
 
-  def link_to_remove_puppetclass(klass, host)
-    options = klass.name.size > 28 ? {:'data-original-title'=>klass.name, :rel=>'twipsy'} : {}
-    functions_options = { :klass => klass, :host => host, :css_class => ''}
-    text = remove_link_to_function(truncate(klass.name, :length => 28), functions_options)
-    content_tag(:span, text, options).html_safe +
-        remove_link_to_function('', functions_options.merge(:css_class => 'glyphicon glyphicon-minus-sign'))
+  def link_to_remove_puppetclass(klass, type)
+    options = options_for_puppetclass_selection(klass, type)
+    text = remove_link_to_function(truncate(klass.name, :length => 28), options)
+    content_tag(:span, text).html_safe +
+        remove_link_to_function('', options.merge!(:class => 'glyphicon glyphicon-minus-sign'))
```

Comment:
Use `merge` and not ! in all of these calls, as the result is being passed into the method.